forgerock-openbanking-uk-aspsp-common/src/test/java/com/forgerock/openbanking/common/services/frequency/FrequencyServiceTest.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright 2019 ForgeRock AS.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.forgerock.openbanking.common.services.frequency;
+
+import com.forgerock.openbanking.common.model.openbanking.frequency.FrequencyType;
+import com.forgerock.openbanking.common.services.openbanking.frequency.FrequencyService;
+import org.joda.time.DateTime;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Unit test for {@link com.forgerock.openbanking.common.services.openbanking.frequency.FrequencyService}.
+ */
+public class FrequencyServiceTest {
+
+    FrequencyService frequencyService;
+    @Before
+    public void setUp() throws Exception {
+        frequencyService = new FrequencyService();
+    }
+
+    // EVERYDAY("EvryDay")
+    @Test
+    public void shouldMatchPattern_everyDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.EVERYDAY.getFrequencyStr();
+
+        // When
+        DateTime dateTime = frequencyService.getNextDateTime(firstDateTime, frequency);
+
+        // Then
+        assertThat(dateTime).isNotNull();
```

Comment:
You could assert a bit more than this - for example:
```
assertThat(dateTime.getDayOfYear()).isEqualTo(firstDateTime.getDayOfYear()+1);
```

Reply:
Isn't possible each test return different nextDateTime, I can check the value changing the private methods on FrequencyService but doesn't make sense.
Now it's a parameterized test and check if exist the nextDateTime.

Reply:
Not even if you set the `DateTime` to a known date, rather than using `DateTime.now()`?

BTW, have you come across `java.time.Clock`? I've used it in the past, but not with Joda.

Reply:
Rollback done.